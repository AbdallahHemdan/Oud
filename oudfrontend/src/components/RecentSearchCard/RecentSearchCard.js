import React, { Component } from "react";
import "./RecentSearchCard.css";
import { Link, withRouter } from "react-router-dom"
import { base, subUrl, prodUrl } from "./../../config/environment"
import PropTypes from "prop-types";
/**
 * Recent Search card component which render and display the playlist card of a specific category 
 * @author Abdallah Hemdan
 * @component
 */

class RecentSearchCard extends Component {
  constructor(props) {
    super(props)
    this.state = {
      _id: "",
      displayName: "",
      type: "",
      images: ["https://oud-zerobase.me/api/uploads/users/default-Profile.svg"]
    }
  }

  /**
   * Function to handle navigation to the playlist page
   * on clicking on the Recent Search card
   * 
   * @function
   * 
   * @return {void}
   * 
   */
  handlePlaylistClick = () => {
    this.props.history.push(`${this.state.type}/${this.state._id}`);
  }

  handleStoringItems = ({ _id, displayName, type, images }) => {
    if (_id !== undefined) {
      this.setState({ _id, displayName, type, images });
    }
  }
  /**
   * Function to handle playing Recent Search on clicking on
   * play icon the Recent Search card
   * 
   * @function
   * 
   * @param {object} event - an event to use it in
   * disabling the default of the propagation 
   */
  handlePlayClick = (e) => {
    e.stopPropagation();
    console.log("ðŸŽµ Recent Search is playing now");
  }
  componentDidMount() {
    const SearchItem = (this.props.item) ? (this.props.item) : null;
    if (SearchItem !== null) {
      this.handleStoringItems(SearchItem);
    }
  }
  /**
   * @function
   * 
   * @name render
   * 
   * @description Render Recent Search card components..
   * 
   * @returns {JSX} Component for App
   */
  render() {
    const subPath = (base === prodUrl) ? subUrl : "";
    return (
      <React.Fragment>
        {
          < div
            className="card-container"
            data-testid="card-container"
          >
            <div className="card"
              data-testid="card"
            >
              <div
                className="overlayer"
                onClick={this.handlePlaylistClick}
                data-testid="overlayer"
              >
                {
                  < button
                    className="play-btn"
                    onClick={this.handlePlayClick}
                    data-testid="play-btn"
                  >
                    <i
                      className="fa fa-play-circle play-circle"
                      data-testid="play-circle"
                    >
                    </i>
                  </button>
                }
              </div>
              <img
                src={`${subPath}${this.state.images[0]}`}
                alt="playlist cover"
                data-testid="playlist-image"
              />
              <div
                className="title"
                data-testid="playlist-title"
              >
                <Link
                  to={`${this.state.type}/${this.state._id}`}
                  className="playlist-link"
                  data-testid="playlist-link"
                >{this.state.displayName}</Link>
              </div>
            </div>
          </ div>
        }
      </React.Fragment>
    );
  }
}
RecentSearchCard.propTypes = {
  item: PropTypes.object
}
export default withRouter(RecentSearchCard);
