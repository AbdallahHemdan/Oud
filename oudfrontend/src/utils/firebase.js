import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
