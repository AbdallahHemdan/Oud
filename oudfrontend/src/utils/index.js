import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
