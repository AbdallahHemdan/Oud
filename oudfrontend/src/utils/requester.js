import axios from "axios";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGFmYTA2NDVmNDU3MTYwNzVmZiIsImlhdCI6MTU4Njg5MzE0MywiZXhwIjoxNTg5NDg1MTQzfQ.ON2Ef2vgOV1_6EokwvD3mlUzgAn0pb5WPCy5qBWj2QA`,
  },
};
const getRequest = (endpoint) => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = (endpoint) => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
