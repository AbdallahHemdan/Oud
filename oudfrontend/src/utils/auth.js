import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
