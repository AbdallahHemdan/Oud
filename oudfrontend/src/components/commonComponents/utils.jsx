import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}