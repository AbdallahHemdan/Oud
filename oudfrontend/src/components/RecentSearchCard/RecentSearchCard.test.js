import React from "react";
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import { findByTestAttr } from "./../../utils/index"
import RecentSearchCard from './RecentSearchCard';
import checkPropTypes from 'check-prop-types'
import { MemoryRouter } from "react-router-dom"
const setup = (props = {}) => {
  const component = shallow(
    <RecentSearchCard.WrappedComponent {...props} />
  )
  return component;
}
const item = {
  _id: 1,
  displayName: "Recently played",
  type: "Artist",
  images: [
    "https://previews.123rf.com/images/aratehortua/aratehortua2001/aratehortua200100022/138417489-vector-cartoon-low-poly-man-with-headphones-character.jpg"
  ]
}
describe('RecentSearch Card component', () => {
  let component;
  beforeEach(() => {
    const props = {
      item
    }
    component = setup(props);
  })
  it('Should render card-container in right way', () => {
    const wrapper = findByTestAttr(component, "card-container");
    expect(wrapper.length).toBe(1);
  });
  it('Should render card-container in right way', () => {
    const wrapper = findByTestAttr(component, "card");
    expect(wrapper.length).toBe(1);
  });
  it('Should render over layer in right way', () => {
    const wrapper = findByTestAttr(component, "overlayer");
    expect(wrapper.length).toBe(1);
  });
  it('Should render play btn in right way', () => {
    const wrapper = findByTestAttr(component, "play-btn");
    expect(wrapper.length).toBe(1);
  });
  it('Should render play circle in right way', () => {
    const wrapper = findByTestAttr(component, "play-circle");
    expect(wrapper.length).toBe(1);
  });
  it('Should render playlist image in right way', () => {
    const wrapper = findByTestAttr(component, "playlist-image");
    expect(wrapper.length).toBe(1);
  });
  it('Should render playlist title in right way', () => {
    const wrapper = findByTestAttr(component, "playlist-title");
    expect(wrapper.length).toBe(1);
  });
  it('Should render playlist link in right way', () => {
    const wrapper = findByTestAttr(component, "playlist-link");
    expect(wrapper.length).toBe(1);
  });
  it('Should render playlist inner text in right way', () => {
    const wrapper = findByTestAttr(component, "playlist-link");
    expect(wrapper.text()).toBe(item.displayName);
  });
  describe('snapshot test', () => {
    it('renders component correctly', () => {
      const wrapper = mount(
        <MemoryRouter>
          <RecentSearchCard.WrappedComponent item={item} />
        </MemoryRouter >
      )
      expect(wrapper).toMatchSnapshot()
    });
  });
  describe('testing prop types', () => {
    it('should pass true props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { ...item }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeUndefined();
    });
    it('should pass false props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { item: 0 }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeDefined();
    });
    it('should pass false props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { item: true }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeDefined();
    });
    it('should pass false props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { item: jest.fn }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeDefined();
    });
    it('should pass false props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { item: [15] }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeDefined();
    });
    it('should pass false props', () => {
      const result = checkPropTypes(RecentSearchCard.WrappedComponent.propTypes, { item: "" }, 'prop', RecentSearchCard.WrappedComponent.name);
      expect(result).toBeDefined();
    });
  })
});