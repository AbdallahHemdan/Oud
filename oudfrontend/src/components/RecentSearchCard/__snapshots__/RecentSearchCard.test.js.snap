// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentSearch Card component snapshot test renders component correctly 1`] = `ReactWrapper {}`;
